test/Unit/models/Import/Service/SendCalculatedResultServiceTest.php

```diff
@@ -0,0 +1,199 @@
+        $this->assertSame(ScoreInterface::GRADING_PROGRESS_STATUS_PENDING_MANUAL, $return['gradingStatus']);
+    }
+
+    public function testDeclarationIsNotScoredVariableIsGraded()
+    {
+        $qtiTestItems = $this->createDeclarations(false);
+
+        $outcomeVariables = $this->createVariables(true);
+
+        $return = $this->doDeliveryExecution($outcomeVariables, $qtiTestItems);
+
+        $this->assertIsArray($return);
+        $this->assertArrayHasKey('gradingStatus', $return);
+        $this->assertSame(ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED, $return['gradingStatus']);
+    }
+
+    public function testDeclarationIsNotScoredVariableNotGraded()
+    {
+        $qtiTestItems = $this->createDeclarations(false);
+
+        $outcomeVariables = $this->createVariables(false);
+
+        $return = $this->doDeliveryExecution($outcomeVariables, $qtiTestItems);
+
+        $this->assertIsArray($return);
+        $this->assertArrayHasKey('gradingStatus', $return);
+        $this->assertSame(ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED, $return['gradingStatus']);
+    }
+
+    public function testDeclarationIsScoredVariableIsGraded()
+    {
+        $qtiTestItems = $this->createDeclarations(true);
+
+        $outcomeVariables = $this->createVariables(true);
+
+        $return = $this->doDeliveryExecution($outcomeVariables, $qtiTestItems);
+
+        $this->assertIsArray($return);
+        $this->assertArrayHasKey('gradingStatus', $return);
+        $this->assertSame(ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED, $return['gradingStatus']);
+    }
+
+    private function createVariables(bool $isExternallyGraded): array
+    {
+        $data = [
+            'type' => taoResultServer_models_classes_OutcomeVariable::TYPE,
+            'normalMaximum' => null,
+            'normalMinimum' => null,
+            'value' => 'MA==',
+            'identifier' => 'OUTCOME_1',
+            'cardinality' => 'single',
+            'baseType' => 'float',
+            'epoch' => '0.80757100 1681904685',
+            'externallyGraded' => $isExternallyGraded,
+        ];
+        $variable = taoResultServer_models_classes_OutcomeVariable::fromData($data);
+
+        $container = new stdClass;
+        $container->variable = $variable;
+
+        return [
+            [
+                $container
+            ]
+        ];
+    }
+
+    private function createDeclarations($isExternallyScored): array
+    {
+        return [
+            'testPart-1' => [
+                'assessmentSection-1' => [
+                    'item-1' => [
+                        'outcomes' => [
+                            [
+                                'identifier' => 'OUTCOME_1',
+                                'attributes' =>
+                                    [
+                                        'identifier' => 'OUTCOME_1',
+                                        'externalScored' => 'externalMachine',
```

Comment:
could you add more complex case, when multiple outcomes declared to be externalMachine/human graded, but only one of corresponding variables got `externallyGraded` **true**?

Reply:
I've added mutliple items and variables to the test